src/electron/platform/android/appium-service-configurator.ts

```diff
@@ -83,13 +90,41 @@ export class AppiumServiceConfigurator implements AndroidServiceConfigurator {
     };
 
     public setTcpForwarding = async (deviceId: string): Promise<number> => {
+        const hostPort = await this.portFinder({
+            port: servicePortNumber,
+            stopPort: servicePortNumber + 100,
+        });
+
         this.adb.setDeviceId(deviceId);
-        await this.adb.forwardPort(defaultAdbPortNumber, defaultAdbPortNumber);
-        return defaultAdbPortNumber;
+        await this.adb.forwardPort(hostPort, servicePortNumber);
```

Comment:
If I'm reading this correctly, it looks like we're going to assign a new port each time we execute this step. If someone were to go several times through the UX via the cancel button, that could accumulate a lot of ports. I wonder if we ought to limit ourselves to only using 1 port at a time--maybe remove a previously forwarded port (within this instance) before forwarding another one? That might also allow us to be less aggressive when removing forwarded ports

Reply:
Agreed